pyupgrade.py

```diff
@@ -1520,12 +1515,15 @@ def visit_If(self, node):  # type: (ast.If) -> None
 
     def visit_For(self, node):  # type: (ast.For) -> None
         if (
-            not self._in_async_def and
+            self._current_sync_def is not None and
             len(node.body) == 1 and
             isinstance(node.body[0], ast.Expr) and
             isinstance(node.body[0].value, ast.Yield) and
             node.body[0].value.value is not None and
             targets_same(node.target, node.body[0].value.value) and
+            not targets_referenced_after_node(
+                self._current_sync_def, node, node.target,
+            ) and
```

Comment:
not a big deal but if you could shorten this to fit on one line that'd be great

Reply:
The naming and signature are okay?

Reply:
yeah I meant if you can come up with more succinct names such that it fits, not a big deal

Reply:
Addressed in 81dc036f6a52e32a3dd1e992334c24fc09cea12b.